snapcraft/_options.py

```diff
@@ -150,6 +150,10 @@ def is_cross_compiling(self):
         return self.__target_machine != self.__platform_arch
 
     @property
+    def target_arch_specified(self):
+        return self.__target_arch_specified
```

Comment:
I didn't go for just saving because `deb_arch` right now means either client or target arch. No strong opinion, though, so I replaced `target_arch_specified` with `target_arch` now which can be **None**.